### PR TITLE
[DO NOT MERGE] Add ecommerce tracking to the new results grouping checker actions

### PR DESCRIPTION
**Pull request by vanitabarrett**
Trello: https://trello.com/c/G8vx3tIU/275-do-not-deploy-adapt-ecommerce-tracking-for-new-results-grouping

## What
Adjust the ecommerce tracking (introduced in https://github.com/alphagov/finder-frontend/pull/1764) for the new groupings.

This is an example of what is sent to GA for an impression and a click:

```
ga("ec:addImpression", {position: 1, list: "Brexit checker results: Visiting the EU", dimension71: "", name: "https://www.gov.uk/guidance/foreign-travel-insurance"})
```

```
ga("ec:addProduct", {position: 2, list: "Brexit checker results: Visiting the EU", dimension71: "", name: "https://www.gov.uk/check-a-passport-for-travel-to-europe"})
```

```
ga("ec:setAction", "click", {list: "Brexit checker results: Visiting the EU"})
```

```
ga("send", {hitType: "event", eventCategory: "UX", eventAction: "click", eventLabel: "Results", dimension15: "200", dimension16: "unknown", dimension95: "1682487469.1574251572", dimension11: "2", dimension3: "other", dimension4: "00000000-0000-0000-0000-000000000000", dimension5: "14", dimension12: "not withdrawn", dimension20: "FinderFrontend", dimension30: "none", dimension32: "none", dimension56: "other", dimension57: "other", dimension58: "other", dimension59: "other", dimension39: "false", dimension26: "0", dimension27: "0", dimension23: "unknown"})
```

**Link to test:**
https://finder-frontend-pr-1752.herokuapp.com/get-ready-brexit-check/results?c%5B%5D=pharma&c%5B%5D=do-not-ip&c%5B%5D=do-not-sell-to-public-sector&c%5B%5D=do-not-eu-uk-funding&c%5B%5D=do-not-personal-eu-org&c%5B%5D=employ-eu-citizens&c%5B%5D=owns-operates-business-organisation&c%5B%5D=visiting-driving&c%5B%5D=visiting-ie&c%5B%5D=visiting-eu&c%5B%5D=visiting-row&c%5B%5D=travel-eu-business-no&c%5B%5D=working-uk&c%5B%5D=living-uk&c%5B%5D=nationality-uk

### CODE REVIEW

File: app/views/brexit_checker/_action_audiences.html.erb [hunk 1]
```diff
@@ -9,7 +9,8 @@
             </h2>
           </header>
           <% audience[:groups].each.with_index do |group, group_index| %>
-            <section class="brexit-checker-actions__group">
+            <% ecommerce_list = group[:heading].nil? ? audience[:heading] : group[:heading] %>
+            <section class="brexit-checker-actions__group" data-analytics-ecommerce data-ecommerce-start-index="1" data-list-title="Brexit checker results: <%= ecommerce_list %>" data-search-query>
```
Comment (huwd) [inline app/views/brexit_checker/_action_audiences.html.erb]: interesting what does that `data-search-query` do?

Comment (vanitabarrett) [reply app/views/brexit_checker/_action_audiences.html.erb]: @huwd It's actually designed for ecommerce tracking on our search pages, and when used there it stores the query that the user has typed in ([see code here](https://github.com/alphagov/static/blob/master/app/assets/javascripts/analytics/ecommerce.js#L13)). Annoyingly the way the code has been written means that it is required when setting up ecommerce tracking, even outside of search pages. We could change this, but it'd be a separate piece of work.

**Comment by vanitabarrett** [comment]
@huwd Oh, doh, I totally forgot this was against the new grouping. I'll raise a PR as we actually want this change to be effective immediately. Then I can rebase this one in preparation for the grouping going live.

**Comment by huwd** [comment]
ah cool cool, peril of this results page hanging around for longer than usual!

We could wait into your other one has gone into master then repoint this at the `groupActionsByResults` [branch or this PR](https://github.com/alphagov/finder-frontend/pull/1731), could merge immediately there so this goes live with the results?

**Comment by vanitabarrett** [comment]
I'm going to close this for now as it's got a bit messy in the rebase. Will re-open a new PR.

**Pull request closed without merging**